sdk/odata/odata.go

```diff
@@ -131,5 +131,17 @@ func (o *OData) UnmarshalJSON(data []byte) error {
 		}
 	}
 
+	for _, k := range []string{"nextLink", "@odata.nextlink"} {
```

Comment:
There's a number of possible values for this field depending on the part of the API in question, so these two fields aren't necessarily sufficient.

Instead the SDK supports [defining a custom pager](https://github.com/hashicorp/go-azure-sdk/blob/8ee56b008ba9275ff6c3282052e6b50e83073f9a/sdk/client/request_options.go#L25-L27) which, given that Pandora has the JSONName for the field containing the pagination details, would allow us to output a `CustomPager` ([like so](https://github.com/hashicorp/go-azure-sdk/blob/8ee56b008ba9275ff6c3282052e6b50e83073f9a/sdk/client/client_test.go#L161-L178)) which is then configured [on the `client.RequestObject` struct](https://github.com/hashicorp/go-azure-sdk/blob/8ee56b008ba9275ff6c3282052e6b50e83073f9a/sdk/client/request_options.go#L25-L27) and ultimately used [when the SDK method definition is output](https://github.com/hashicorp/go-azure-sdk/blob/8ee56b008ba9275ff6c3282052e6b50e83073f9a/resource-manager/apimanagement/2023-05-01-preview/apiissue/method_get.go#L50-L58).

We're currently using this information [to build out the AutoRest methods that use polling](https://github.com/hashicorp/pandora/blob/7b47fa7e752c27291ec8cb983e4bfb3005a31a5a/tools/generator-go-sdk/internal/generator/templater_methods_autorest.go#L610) so once the `CustomPoller` is output, that should be all that's needed.

As such, rather than doing this here - can we update [`generator-go-sdk` to output a `CustomPoller` as needed here](https://github.com/hashicorp/pandora/blob/7b47fa7e752c27291ec8cb983e4bfb3005a31a5a/tools/generator-go-sdk/internal/generator/templater_methods.go#L505-L514) instead, which'll fix this regardless of the field type?